app/models/shipit/rollback.rb

```diff
@@ -33,13 +33,18 @@ def to_partial_path
       'deploys/deploy'
     end
 
+    def report_complete!
+      complete!
+    end
+
     private
 
     def update_release_status
       return unless stack.release_status?
 
       # When we rollback to a certain revision, assume that all later deploys were faulty
       stack.deploys.newer_than(deploy.id).until(stack.last_completed_deploy.id).to_a.each do |deploy|
+        next if deploy.id == id
```

Comment:
I don't get it, `newer_then(deploy.id)` should translate as `id > #{deploy.id}`, so `deploy` shouldn't be yielded by that `each`.